lib/galaxy/managers/job_connections.py

```diff
@@ -54,7 +54,8 @@ def get_related_hids(self, history_id, hid):
             for val in graph["outputs"] + graph["inputs"]:
                 item_class = get_class(val["src"])
                 item = self.sa_session.query(item_class).get(val["id"])
```

Comment:
That also results in an warning that you're using `None` as the key. I think you need to filter higher up in get_connections_graph

Reply:
Ah, by using `None` as the key, do you mean `val[id]` or `val[src]` could be `None`, so I should make sure that if that is the case, I shouldn’t even add an item to the graph in `get_connections_graph`?

Reply:
Or rather, I shouldn’t even add an item to the result in `_get_union_results`

Reply:
Right, you can filter for JobToInputDatasetAssociation.dataset_id / JobToInputDatasetCollectionAssociation.dataset_collection_id to not be null. I'd also do `hid = self.sa_session.query(item_class.hid).get(val["id"])`, that means you're not pulling in all those attributes you're never using. 

Longer term I'd pull this all into a single query, but that can wait.

Reply:
>  I'd also do hid = self.sa_session.query(item_class.hid).get(val["id"])

That's a great idea, I'll change it to that
Filtering `_id != null` makes a lot more sense too, thanks! Will update with these changes

Reply:
`self.sa_session.query(item_class.hid).get(val["id"])` wouldn't work because:
error = `sqlalchemy.exc.InvalidRequestError: get() can only be used against a single mapped class.`

So, I did:`self.sa_session.execute(select(item_class.hid).where(item_class.id == val["id"])).scalar()` instead